app/src/org/commcare/android/database/DbHelper.java

```diff
@@ -33,7 +34,7 @@ public DbHelper(Context c) {
         this.c = c;
     }
     
-    public abstract SQLiteDatabase getHandle();
+    public abstract SQLiteDatabase getHandle() throws SessionUnavailableException;
```

Comment:
Instead of throwing a null pointer exception when the user database is closed I think it makes best sense to throw a custom checked exception. This way we are forced to handle the session expiring and closing down the DB instead of letting the runtime exception crash the App.

If you don't agree with this design decision then this whole PR is probably moot.